utils/frame/try-runtime/cli/src/lib.rs

```diff
@@ -111,6 +103,19 @@ pub struct SharedParams {
 	/// sc_service::Configuration.default_heap_pages.
 	#[structopt(long)]
 	pub heap_pages: Option<u64>,
+
+	/// The block hash at which to read state. This is required for execute-block, offchain-worker,
+	/// or any command that used the live subcommand.
+	#[structopt(
+		short,
+		long,
+		multiple = false,
+		parse(try_from_str = parse::hash),
+		required_ifs(
+			&[("command", "offchain-worker"), ("command", "execute-block"), ("subcommand", "live")]
+		)
+	)]
+	pub block_at: String,
```

Comment:
It looks like every time you use `shared.block_at`, the pattern is `shared.block_at.parse().map_err(...)?`. Could you make the struct member here a `Block::Hash` and update the `parse::hash` command to emit that type?

Reply:
@coriolinus I have been trying to switch the `block_at` struct member type to the associated type `Block::Hash`, but I have been getting a bit turned around with the compiler errors. I here is a diff that shows what I am trying to do: https://github.com/paritytech/substrate/compare/7c2e3515095de6bf094673f0a0c6d82192532a79..c4138829554952334dac679e93152cb72089e308

Reply:
Ah, I see the issue. `Block` is still generic at that point, and as your diff shows, it's hard to reify it that early. 

What do you think of the approach in #9153?

Reply:
Great thank you! Merged